src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -162,6 +162,11 @@ def resolve(self, root_reqs, check_supported_wheels):
 
         reqs = req_set.all_requirements
         self.factory.preparer.prepare_linked_requirements_more(reqs)
+
+        # TODO: extricate this call from the resolver.resolve() code path once
+        # we can drop the v1 resolver.
+        self.factory.preparer.complete_partial_requirements(req_set)
```

Comment:
Hmm... Since this is gonna be a TODO anyway, maybe we should make this call within preparer itself?